omod/src/main/java/org/openmrs/module/stockmanagement/web/controller/StockManagementRestController.java

```diff
@@ -0,0 +1,117 @@
+import org.openmrs.api.PersonService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+// import org.openmrs.ui.framework.SimpleObject;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.openmrs.module.stockmanagement.api.Privileges;
+import org.openmrs.module.stockmanagement.api.StockManagementService;
+import org.openmrs.module.stockmanagement.api.dto.RecordPrivilegeFilter;
+import org.openmrs.module.stockmanagement.api.dto.StockInventoryResult;
+import org.openmrs.module.stockmanagement.api.dto.StockItemInventory;
+import org.openmrs.module.stockmanagement.api.dto.StockItemInventorySearchFilter;
+import org.openmrs.module.stockmanagement.api.utils.SimpleObject;
+
+import org.springframework.web.bind.annotation.CrossOrigin;
+
+/**
+ * This class configured as controller using annotation
+ */
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/stockmanagement/metrics")
+@Authorized
+public class StockManagementRestController {
+	
+	/** Logger for this class and subclasses */
+	protected final Log log = LogFactory.getLog(getClass());
+	
+	/**
+     * end point for getting metrics for total out of stock items
+     * @param request
+     * @return
+     */
+    @CrossOrigin(origins = "*", methods = {RequestMethod.GET, RequestMethod.POST, RequestMethod.PUT, RequestMethod.DELETE, RequestMethod.OPTIONS})
+    @RequestMapping(method = RequestMethod.GET, value = "/outofstockitemstotal") 
+    @ResponseBody
+    public Object getOutOfStockItemTotal(HttpServletRequest request) {
+		SimpleObject ret = new SimpleObject();
+        List<SimpleObject> results = new ArrayList<>();
+
+		try {
+			StockManagementService stockManagementService = Context.getService(StockManagementService.class);
+			results = stockManagementService.getOutOfStockItemsMetrics();
+		} catch(Exception ex) {
+			ex.printStackTrace();
```

Comment:
Ideally, the end point should report 500 error if it cannot get results. This makes it look like there are just no items.